File: src/analysis/ParticleRadiusDistribution.py
"""
************************************************
**espresso.analysis.ParticleRadiusDistribution**
************************************************

"""
from espresso.esutil import cxxinit
from espresso import pmi

from espresso.analysis.AnalysisBase import *
from _espresso import analysis_ParticleRadiusDistribution

class ParticleRadiusDistributionLocal(AnalysisBaseLocal, analysis_ParticleRadiusDistribution):
    'The (local) compute of the particle radius distribution.'
    def __init__(self, system):
        if not pmi._PMIComm or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, analysis_ParticleRadiusDistribution, system)

if pmi.isController :
    class ParticleRadiusDistribution(AnalysisBase):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espresso.analysis.ParticleRadiusDistributionLocal'
            )

File: src/analysis/ParticleRadiusDistribution.cpp
#include "python.hpp"
#include "ParticleRadiusDistribution.hpp"

namespace espresso {
  namespace analysis {

    void ParticleRadiusDistribution::registerPython() {
      using namespace espresso::python;
      class_<ParticleRadiusDistribution, bases< AnalysisBase > >
        ("analysis_ParticleRadiusDistribution", init< shared_ptr< System > >())
      ;
    }
  }
}
